File: WWAlertView/WWAlertView/WWTextField.m
//
//  WWTextField.m
//  WWAlertView
//
//  Created by XingLei on 15/12/11.
//  Copyright © 2015年 XingLei. All rights reserved.
//

#import "WWTextField.h"
#import "ControlColor.h"

#define lineHeight          3

@implementation WWTextField

#pragma mark -
#pragma mark Init

- (instancetype)initWithFrame:(CGRect)frame {
    self = [super initWithFrame:frame];
    if (!self) return nil;

    self.textColor = [UIColor grayColor];
    self.font = [UIFont systemFontOfSize:15.0f];
    self.placeholder = @"请输入验证码";
    self.layer.cornerRadius = 4;
    self.layer.borderColor = [ControlColor colorWithHexString:@"#cccccc"].CGColor;
    self.layer.borderWidth = 1.0f;
    self.backgroundColor = [ControlColor colorWithHexString:@"#f5f5f5"];
    self.keyboardType = UIKeyboardTypeNumberPad;
    [self addLineView];
    return self;
}

- (void)addLineView {
    UIGraphicsBeginImageContextWithOptions(self.frame.size, 0, 0);
    [[ControlColor colorWithHexString:@"#cccccc"]setFill];
    UIBezierPath *path = [UIBezierPath bezierPath];
    [path moveToPoint:CGPointMake(self.frame.size.width * 0.65, lineHeight + 2)];
    [path addLineToPoint:CGPointMake(self.frame.size.width * 0.65, lineHeight + (self.frame.size.height - lineHeight*lineHeight))];
    path.lineWidth = 0.3;
    [path stroke];
    UIImage *image = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    
    UIImageView *imageView = [[UIImageView alloc]initWithImage:image];
    [self addSubview:imageView];
}

- (CGRect)textRectForBounds:(CGRect)bounds {
    return CGRectInset( bounds , 10 , 0 );
}

- (CGRect)editingRectForBounds:(CGRect)bounds {
    return CGRectInset( bounds , 10 , 0 );
}

- (void)drawPlaceholderInRect:(CGRect)rect
{
    [[UIColor grayColor] setFill];
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wdeprecated-declarations"
    [[self placeholder] drawInRect:rect withFont:[UIFont systemFontOfSize:15.0f]];
#pragma clang diagnostic pop
}

- (CGRect)placeholderRectForBounds:(CGRect)bounds
{
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wdeprecated-declarations"
    CGSize size = [self.placeholder sizeWithFont:[UIFont systemFontOfSize:15.0f]];
#pragma clang diagnostic pop
    CGRect placeHolderF = CGRectMake(10, 0, size.width, size.height);
    return placeHolderF;
}

@end

File: WWAlerViewController_Swift/WWAlerViewController_Swift/WWTextField.swift
//
//  WWTextField.swift
//  WWAlerViewController_Swift
//
//  Created by XingLei on 15/12/15.
//  Copyright © 2015年 XingLei. All rights reserved.
//

import UIKit

let lineHeight: CGFloat =  3.0
let lineOrigin: CGFloat = 0.65

class WWTextField: UITextField {
    
    var layerColor = UIColor()
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        textColor = UIColor.blackColor()
        font = UIFont.systemFontOfSize(12)
        placeholder = "请输入验证码"
        layer.cornerRadius = 4
        layer.borderColor = layerColor.CGColor
        layer.borderWidth = 1.0
        backgroundColor = UIColor.blackColor()
        keyboardType = UIKeyboardType.NumbersAndPunctuation
    }
    
    func addLineView() {
        UIGraphicsBeginImageContextWithOptions(frame.size, false, 0)
        UIColor.blackColor().setFill()
        let bezier = UIBezierPath()
        bezier.moveToPoint(CGPointMake(frame.size.width * lineOrigin, lineHeight + 2))
        bezier.addLineToPoint(CGPointMake(frame.size.width * lineOrigin, lineHeight + (frame.height - lineHeight*lineHeight)))
        bezier.lineWidth = 0.3
        bezier.stroke()
        let image = UIGraphicsGetImageFromCurrentImageContext()
        let imageView = UIImageView(image: image)
        addSubview(imageView)
    }
    
    override func drawPlaceholderInRect(rect: CGRect) {
        UIColor.grayColor().setFill()
        let topStringFont = UIFont().fontWithSize(15)
        let attributesDictionary = [NSFontAttributeName: topStringFont] as NSDictionary
        placeholder?.drawInRect(rect, withAttributes:attributesDictionary as? [String : AnyObject])
    }
    
    override func placeholderRectForBounds(bounds: CGRect) -> CGRect {
        let attributesDictionary = [NSFontAttributeName: UIFont().fontWithSize(15)]
        let placeHolderSize: CGSize = placeholder!.sizeWithAttributes(attributesDictionary)
        let placeHolderF: CGRect = CGRectMake(10, 1, placeHolderSize.width, placeHolderSize.height)
        
        return placeHolderF
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
}
